# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1255691616 -7200
#      Fri Oct 16 13:13:36 2009 +0200
# Node ID aab4f2aa9ed9bafe0763fb5615d6cb56d354537e
# Parent  b4fdfee405b5cab18551b0fb6705bf7b300f1da5
make ismember more matlab compatible

diff --git a/scripts/set/ismember.m b/scripts/set/ismember.m
--- a/scripts/set/ismember.m
+++ b/scripts/set/ismember.m
@@ -128,30 +128,34 @@ function [tf, a_idx] = ismember (a, s, r
   elseif (nargin == 3 && strcmpi (rows_opt, "rows"))
     if (iscell (a) || iscell (s))
       error ("ismember: cells not supported with ""rows""");
     elseif (! isa (a, class (s))) 
       error ("ismember: both input arguments must be the same type");
     elseif (! ischar (a) && ! isnumeric (a))
       error ("ismember: input arguments must be arrays, cell arrays, or strings"); 
     endif
-    if (columns (a) != columns (s))
+    if (isempty (a) || isempty (c))
+      tf = false (rows (a), 1);
+      a_idx = zeros (rows (a), 1);
+    elseif (columns (a) != columns (s))
       error ("ismember: number of columns must match");
-    endif
+    else
 
-    ## FIXME: lookup does not support "rows", so we just use unique.
-    [xx, ii, jj] = unique ([a; s], "rows", "last");
-    na = rows (a);
-    jj = ii(jj(1:na));
-    tf = jj > na;
+      ## FIXME: lookup does not support "rows", so we just use unique.
+      [xx, ii, jj] = unique ([a; s], "rows", "last");
+      na = rows (a);
+      jj = ii(jj(1:na));
+      tf = jj > na;
 
-    if (nargout > 1)
-      a_idx = max (0, jj - na);
+      if (nargout > 1)
+        a_idx = max (0, jj - na);
+      endif
+
     endif
-
   else
     print_usage ();
   endif
 
 endfunction
 
 %!assert (ismember ({''}, {'abc', 'def'}), false);
 %!assert (ismember ('abc', {'abc', 'def'}), true);
